frontstage/controllers/party_controller.py

```diff
@@ -796,3 +793,49 @@ def reset_password_reset_counter(party_id):
 
     logger.info("Successfully reset password reset counter")
     return response.json()
+
+
+def get_business_survey_enrolments_map(party_id: str) -> dict:
```

Comment:
This realistically should be done at source (party) but this a big step in the right direction. It replaces the relentless calls to the party service and just uses the single enrolments endpoint and works everything out. 

If you had 10 surveys in your account from you would get (31) calls to party

The current implementation called the respondent endpoint for it's id (1), then looped around each business_id calling party enrolments for each business (10), but only used the id to  calling business again for the details (10) to top it off it also called survey (10)

It now does it in one

Reply:
I think that's fine. I was going to question if the party_controller was the right place for these structure wrangler methods, but as you say this should be done in the party service so confining them here makes sense for the time being.